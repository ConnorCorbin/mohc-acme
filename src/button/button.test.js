import React from 'react';

import Button from 'button/button';

import StyledButton from 'button/styles/button';

describe('Button', () => {
  let wrapper;
  const getWrapper = ({
    buttonText = 'Nice button text',
    type = 'submit',
    isDisabled = true,
    designType = 'submit',
    onClick,
  } = {}) => shallow(
    <Button
      buttonText={buttonText}
      type={type}
      isDisabled={isDisabled}
      designType={designType}
      onClick={onClick}
    />,
  );

  it('should render component', () => {
    wrapper = getWrapper();

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.isEmptyRender()).toEqual(false);
  });

  it('should render component with correct text', () => {
    wrapper = getWrapper();

    expect(wrapper.find(StyledButton).children().text()).toEqual('Nice button text');
  });

  it('should render component with correct prop values', () => {
    wrapper = getWrapper();

    expect(wrapper.find(StyledButton).props()).toMatchObject({
      type: 'submit',
      disabled: true,
      designType: 'submit',
    });
  });

  it('should execute onClickFunction when StyledButton onClick event is fired', () => {
    const mockFunction = jest.fn();
    wrapper = getWrapper({ onClick: mockFunction });

    expect(mockFunction).not.toHaveBeenCalledWith();
    wrapper.find(StyledButton).simulate('click');
    expect(mockFunction).toHaveBeenCalledWith();
  });
});
