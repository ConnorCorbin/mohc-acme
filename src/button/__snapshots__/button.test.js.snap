// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render component 1`] = `
<styled.div>
  <styled.button
    designType="submit"
    disabled={true}
    type="submit"
  >
    Nice button text
  </styled.button>
</styled.div>
`;
